### ISSUE TITLE
[Bug]: Check for determining "project file" does not correctly work for solution filter files

### ISSUE THREAD

**Issue by stefan-schweiger** [dotnet/msbuild#9587]
### Issue Description

When running `dotnet build` in a folder which only contains a `.slnf` the commands won't work because no valid "project file" can be determined, but if you run `dotnet build MyFilter.slnf` the commands correctly work.

### Steps to Reproduce

Create a solution with the following structure and run `dotnet build` in the folder `ServiceA`

```
/
├─ ServiceA/
│  ├─ ServiceA.QueryHandling/
│  │  ├─ ServiceA.QueryHandling.csproj
│  ├─ ServiceA.slnf
├─ MySolution.sln
```

### Expected Behavior

The build should correctly pick up the solution filter file and work like it would in the `/` and `ServiceA.QueryHandling` folders.

### Actual Behavior

If you run it in the `ServiceA` folder you get this error:
`MSB1003: Specify a project or solution file. The current working directory does not contain a project or solution file.`

### Analysis

I think the problem is on line 3422 here that it only looks for `*.sln` files and therefore never picks up `*.slnf` files, so it should probably be changed to `*.sln*` to actually correctly pick up the solution filter files.

https://github.com/dotnet/msbuild/blob/5d663a0cbe85b5b4e70c3a07068e96cfaa5c6a70/src/MSBuild/XMake.cs#L3422-L3497

### Versions & Configurations

MSBuild version 17.8.3+195e7f5a3 for .NET
17.8.3.51904%

dotnet 8.0.100

macOS 14.2.1 (23C71)

**Comment by stefan-schweiger** [comment]
Also I think the "mock" code here

https://github.com/dotnet/msbuild/blob/5d663a0cbe85b5b4e70c3a07068e96cfaa5c6a70/src/MSBuild.UnitTests/XMake_Tests.cs#L1931-L1953

for this test does not correctly work

https://github.com/dotnet/msbuild/blob/5d663a0cbe85b5b4e70c3a07068e96cfaa5c6a70/src/MSBuild.UnitTests/XMake_Tests.cs#L1614-L1625

**Comment by AR-May** [comment]
Team triage: @stefan-schweiger your analysis looks good, if you are interested, please send a PR fixing that.

**Comment by jrdodds** [comment]
This bug has been fixed by Feature Request #10266 and PR #10794. The PR implemented the changes identified by @stefan-schweiger. I confirmed by testing and reviewing the code.

I believe this issue can be closed.

**Comment by rainersigwald** [comment]
Thanks @jrdodds!
